distributed-process-tests/src/Control/Distributed/Process/Tests/CH.hs

```diff
@@ -1369,6 +1372,116 @@ testExitRemote TestTransport{..} = do
   takeMVar supervisedDone
   takeMVar supervisorDone
 
+testRegistryMonitoring :: TestTransport -> Assertion
+testRegistryMonitoring TestTransport{..} = do
+  localNode <- newLocalNode testTransport initRemoteTable
+  remoteNode <- newLocalNode testTransport initRemoteTable
+  return ()
+
+  -- Local process. Test if local process will be removed from
+  -- registry when it dies.
+  box <- newEmptyMVar
+  runProcess localNode $ do
+    pid <- spawnLocal $ do
+                expect
+    register "test" pid
+    tpid <- whereis "test"
+    if tpid == Just pid
+      then do _ <- monitor pid
+              send pid ()
+              ProcessMonitorNotification{} <- expect
+              tpid1 <- whereis "test"
+              liftIO $ putMVar box (Nothing == tpid1)
+      else liftIO $ putMVar box False
+
+  takeMVar box >>= assertBool "expected local process to not be registered"
+  return ()
+
+  -- Remote process. Test if remote process entry is removed
+  -- from registry when process dies.
+  remote1 <- testRemote remoteNode
+  runProcess localNode $
+    let waitpoll = do
+        w <- whereis "test" :: Process (Maybe ProcessId)
+        forM_ w (const waitpoll)
+    in do register "test" remote1
+          -- [race-condition] invalid test if we force exit before waitpoll runs
+          registered <- whereis "test"
+          registered `shouldBe` (equalTo (Just remote1))
+          send remote1 ()
+          waitpoll
+          return ()
+  return ()
+
+  -- Many labels. Test if all labels associated with process
+  -- are removed from registry when it dies.
+  remote2 <- testRemote remoteNode
+  runProcess localNode $
+    let waitpoll = do
+        w1 <- whereis "test-3" :: Process (Maybe ProcessId)
+        w2 <- whereis "test-4" :: Process (Maybe ProcessId)
+        forM_ (w1 <|> w2) (const waitpoll)
+    in do register "test-3" remote2
+          register "test-4" remote2
+          send remote2 ()
+          waitpoll
+          return ()
+
+{- XXX: waiting including patch for nsend for remote process
```

Comment:
I think that the patch has landed, if not, can we add a link to that?

Reply:
I thought so too, just wanted confirmation. I'll uncomment and we'll see what Travis thinks!